packages/flutter/lib/src/rendering/list_wheel_viewport.dart

```diff
@@ -562,34 +626,109 @@ class RenderListWheelViewport
       perspective: _perspective,
     );
 
-    context.pushTransform(
-      // Text with TransformLayers and no cullRects currently have an issue rendering
-      // https://github.com/flutter/flutter/issues/14224.
-      false,
-      offset,
-      _centerOriginTransform(transform),
-      // Pre-transform painting function.
-      (PaintingContext context, Offset offset) {
-        context.paintChild(
-          child,
-          offset + new Offset(
-            untransformedPaintingCoordinates.dx,
-            // Paint everything in the center (e.g. angle = 0), then transform.
-            -_topScrollMarginExtent,
-          ),
-        );
-      },
-    );
+    void _paintCylindrically(PaintingContext context, Offset offset) {
+      context.pushTransform(
+        // Text with TransformLayers and no cullRects currently have an issue rendering
+        // https://github.com/flutter/flutter/issues/14224.
+        false,
+        offset,
+        _centerOriginTransform(transform),
+        // Pre-transform painting function.
+            (PaintingContext context, Offset offset) {
+          context.paintChild(
+            child,
+            offset + new Offset(
+              untransformedPaintingCoordinates.dx,
+              // Paint everything in the center (e.g. angle = 0), then transform.
+              -_topScrollMarginExtent,
+            ),
+          );
+        },
+      );
+    }
+
+    if (!useMagnifier) {
+      _paintCylindrically(context, offset);
+      return;
+    }
+
+    final double magnifierTopLinePosition =
+        size.height / 2 - _itemExtent * _magnifyRate / 2;
+    final double magnifierBottomLinePosition =
+        size.height / 2 + _itemExtent * _magnifyRate / 2;
+
+    // Some part of the child is in the center magnifier.
+    if (untransformedPaintingCoordinates.dy
```

Comment:
Give these 2 conditions a variable with names so it's more readable than re-doing the math in the reader's head.

Reply:
done.